### PR TITLE
Sensitive Value Hints

### PR DESCRIPTION
**Pull request by adam-mccoy**
# Background

The Hint property is intended to allow sensitive values to indicate what their value is. For example, it might provide the first few characters of a authorization token.

### CODE REVIEW

**Comment by adam-mccoy** [comment]
[Display masked API key](https://trello.com/c/iJGDDeQk/398-display-masked-api-key)

File: source/Octopus.Data/Resources/SensitiveValue.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@ public class SensitiveValue : IEquatable<SensitiveValue>
     {
         public bool HasValue { get; set; }
         public string? NewValue { get; set; }
+        public string? Hint { get; set; }
```
Comment (andyinaus) [inline source/Octopus.Data/Resources/SensitiveValue.cs:10]: Wondering if we should compare this property in its `Equals` method or not

Comment (slewis74) [reply source/Octopus.Data/Resources/SensitiveValue.cs:10]: I actually don't know where these would be used, but from a look at the compare methods I think they may be wrong. Or at least very misleading. If I'm reading right, if 2 things that had different values had been loaded they'd both have `HasValue == true` and they would also both have null NewValue. So they would look the same, even though they are not. I think we should actually remove all of this.

**Review by slewis74 — CHANGES_REQUESTED** [review]
The Hint looks good. I think the equality compares should be removed, so as not to be misleading.

**Pull request merged**